ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts/switch.py

```diff
@@ -0,0 +1,856 @@
+
+        switch['bgp_peer_groups'] = get_bgp_peer_groups(vars)
+
+        switch['router_id'] = get_router_id(vars, template_lookup_module)
+
+        # IPv6 related facts
+        if vars.get('underlay_ipv6') is True:
+            switch.update(get_facts(UNDERLAY_ROUTER_IPV6_FACTS, switch_data_combined))
+
+            switch['ipv6_router_id'] = get_ipv6_router_id(vars, template_lookup_module)
+
+        switch['evpn_role'] = default(switch_data_combined.get('evpn_role'), switch['default_evpn_role'])
+
+        switch['mpls_overlay_role'] = default(switch_data_combined.get('mpls_overlay_role'),
+                                              switch['default_mpls_overlay_role'])
+
+        if switch['underlay_routing_protocol'] == 'ebgp' or switch['evpn_role'] != 'none' or switch['mpls_overlay_role'] != 'none':
+            switch['bgp_as'] = get_bgp_as(vars, switch_data_combined)
+        # Hack to make mpls PR non-breaking, adds empty bgp to igp topology spines
+        elif switch['underlay_routing_protocol'] in ['isis', 'ospf'] and switch['evpn_role'] == 'none' and vars.get('bgp_as') is not None:
+            switch['bgp_as'] = str(vars['bgp_as'])
+
+        # For clients the default value for EVPN Route Servers is content of the uplink_switches variable set above.
+        if switch['evpn_role'] == "client":
+            switch['evpn_route_servers'] = default(switch_data_combined.get('evpn_route_servers'), switch['uplink_switches'])
+        else:
+            switch['evpn_route_servers'] = switch_data_combined.get('evpn_route_servers')
+
+        if switch['mpls_overlay_role'] in ["client", "server"]:
+            switch['mpls_route_reflectors'] = switch_data_combined.get('mpls_route_reflectors')
+
+        if switch['underlay_routing_protocol'] in ['isis', 'isis-ldp', 'isis-sr', 'isis-sr-ldp']:
+            switch.update(get_isis_facts(vars, switch_data_combined))
+
+    # #######################
+    # MLAG related facts
+    if switch['mlag_support'] is True and switch_data_combined.get('mlag', True) is True and len(switch_data_node_group_nodes) == 2:
+        switch['mlag'] = True
+
+        switch.update(get_facts(MLAG_FACTS, switch_data_combined))
+
+        if list(switch_data_node_group_nodes.keys()).index(vars['inventory_hostname']) == 0:
+            switch['mlag_role'] = "primary"
+            switch['mlag_peer'] = list(switch_data_node_group_nodes.keys())[1]
+
+        elif list(switch_data_node_group_nodes.keys()).index(vars['inventory_hostname']) == 1:
+            switch['mlag_role'] = "secondary"
+            switch['mlag_peer'] = list(switch_data_node_group_nodes.keys())[0]
+
+        if switch.get('underlay_router') is True:
+            switch['mlag_l3'] = True
+            mlag_peer_vlan = switch['mlag_peer_vlan']
+            if switch_data_combined.get('mlag_peer_l3_vlan', 4093) not in [None, False, mlag_peer_vlan]:
+                switch['mlag_peer_l3_vlan'] = switch_data_combined.get('mlag_peer_l3_vlan', 4093)
```

Comment:
```suggestion
            combined_peer_l3_vlan = switch_data_combined.get('mlag_peer_l3_vlan', 4093)
            if combined_peer_l3_vlan not in [None, False, switch['mlag_peer_vlan']:
                switch['mlag_peer_l3_vlan'] = combined_peer_l3_vlan
```

remove one `get` call